### PR TITLE
Optimizing performance for Pulsar function archive download

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Currently function upload, download, create, and update does really stream the function package to or from the function worker rather it reads all of the data in memory and then uploads or writes to file which can cause memory issues especially if the function package is big

### CODE REVIEW

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java [hunk 1]
```diff
@@ -387,6 +393,14 @@ public void close() {
             LOG.error("Failed to close the authentication service", e);
         }
         client.close();
+
+        if (httpAsyncClient != null) {
```
Comment (merlimat) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java]: This will always be non null

**Comment by jerrypeng** [comment]
rerun integration tests

**Comment by jiazhai** [comment]
run java8 tests
run integration tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Pull request merged**